Firebase/ios/Bonfire/Mobile/Utilities/FirebaseConvertible.swift

```diff
@@ -1,8 +1,20 @@
 import Foundation
+import Firebase
 
 protocol FirebaseConvertible {
+    init(snapshot: FIRDataSnapshot) throws
     init(firebaseValue: AnyObject) throws
     func asFirebaseValue() -> AnyObject
 }
 
+extension FirebaseConvertible {
+    init(snapshot: FIRDataSnapshot) throws {
```

Comment:
This init makes it easier to add the conversion to a typed model object into the rx pipeline. The only caveats is sometimes `FIRDataSnapshot` come out typed as `AnyObject`, which means the `firebaseValue` initialiser gets called instead, so some care is required.